sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java

```diff
@@ -342,6 +342,21 @@ public void testFilteredSelfJoin() throws Exception {
     }
 
     @Test
+    public void testFilteredMatchPredicateSelfJoin() {
+        execute("create table test1(id long primary key, num long, txt string index using fulltext) with (number_of_replicas=10)");
```

Comment:
imho it does not make sense to specify 10 replicas here. this test should run with 2 datanodes (probably in a sperate class if needed) and with 1 replica. also you need to call ensuregreen, only then the shard allocation is different. however i think a unit test would make sense for the routing allocation.